sig-and-wg/sig-core/proposals/update-code-and-chart-in-one-pr.md

```diff
@@ -0,0 +1,86 @@
+# Update component and its chart in one pull request
+
+Created on 2019-23-04 by Adam Szecowka (@aszecowka) and Pawel Kosiec (@pkosiec). 
+
+## Status
+Proposed on 2019-23-04.
+
+## Motivation
+Currently, when a developer introduces changes to his component, he needs to do at least 2 PR:
+
+- Changes made to the component
+- Bump image version in the chart
+
+Such an approach has many drawbacks:
+
+- Slow development cycle, dealing with 2 PRs, asking for approvals twice
+- A developer can forget to bump image and version can be updated only on the release day. Updating version on the release day is 
+the worst case scenario because we postponed integration of our components. This is against Continuous Integration approach.
+- The code in the repository does not reflect what is really running.
+- Ugly repository history. Many commits only bump the component's image version.
+
+## Goal
+- Kyma developer can introduce his changes to Kyma in one Pull Request. Changes to the component and helm chart where we update component version is done in the same PR.
+- The code in the Kyma repository reflects code executed on the Kyma.
+When doing a release we rebuild all components. We should not be surprised on the release day that some tests are failing because some component was not updated for 3 weeks
```

Comment:
this changes the meaning of the sentence: In my sentence, it can happen (let assume 20% chances), in your sentence, it is going to happen (100% chances).